server/src/main/java/org/opensearch/search/aggregations/AggregatorFactory.java

```diff
@@ -114,4 +114,17 @@ public AggregatorFactory getParent() {
     public String getStatsSubtype() {
         return OTHER_SUBTYPE;
     }
+
+    protected boolean supportsConcurrentSegmentSearch() {
+        return false;
+    }
+
+    public boolean traverseChildFactories() {
+        for (AggregatorFactory aggregatorFactory : factories.getFactories()) {
+            if (aggregatorFactory.traverseChildFactories() == false) {
+                return false;
+            }
+        }
+        return supportsConcurrentSegmentSearch();
```

Comment:
we can remove this and in the override of `supportsConcurrentSegmentSearch` each `AggregationFactory` that supports it will just call this method as suggested below.

Reply:
I like the suggestion above of `allFactoriesSupportConcurrentSearch` so we can move the iteration logic out of `DefaultSearchContext`, but I think the pattern of `supportsConcurrentSegmentSearch` being the recursive base case is a lot more intuitive -- when reading the code seeing `supportsConcurrentSegmentSearch()` returning `true` if it supports concurrent search is a lot more intuitive than returning `traverseChildFactories()`.

Reply:
I moved the `factories`logic out of `DefaultSearchContext` and now all of the iteration will happen in `AggregatorFactories` while `AggregatorFactory` will just perform the recursive call.